server/sonar-server/src/main/java/org/sonar/server/organization/ws/CreateAction.java

```diff
@@ -0,0 +1,203 @@
+  private static final String PARAM_KEY = "key";
+  private static final String PARAM_DESCRIPTION = "description";
+  private static final String PARAM_URL = "url";
+  private static final String PARAM_AVATAR_URL = "avatar";
+  private static final int NAME_MIN_LENGTH = 2;
+  private static final int NAME_MAX_LENGTH = 64;
+  private static final int KEY_MIN_LENGTH = 2;
+  private static final int KEY_MAX_LENGTH = 32;
+
+  private final UserSession userSession;
+  private final DbClient dbClient;
+  private final UuidFactory uuidFactory;
+  private final System2 system2;
+
+  public CreateAction(UserSession userSession, DbClient dbClient, UuidFactory uuidFactory, System2 system2) {
+    this.userSession = userSession;
+    this.dbClient = dbClient;
+    this.uuidFactory = uuidFactory;
+    this.system2 = system2;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION)
+      .setPost(true)
+      .setDescription("Create an organization.<br /> Requires 'Administer System' permission.")
+      .setResponseExample(getClass().getResource("example-create.json"))
+      .setInternal(true)
+      .setSince("6.2")
+      .setHandler(this);
+
+    action.createParam(PARAM_NAME)
+      .setRequired(true)
+      .setDescription("Name of the organization. <br />" +
+        "It must be between 2 and 64 chars longs.")
+      .setExampleValue("Foo Company");
+
+    action.createParam(PARAM_KEY)
+      .setRequired(false)
+      .setDescription("Key of the organization. <br />" +
+        "The key is unique to the whole SonarQube. <br/>" +
+        "When not specified, the key is computed from the name. <br />" +
+        "Otherwise, it must be between 2 and 32 chars long. All chars must be lower-case letters (a to z), digits or dash (but dash can neither be trailing nor heading)")
+      .setExampleValue("foo-company");
+
+    action.createParam(PARAM_DESCRIPTION)
+      .setRequired(false)
+      .setDescription("Description of the organization.<br/> It must be less than 256 chars long.")
+      .setExampleValue("The Foo company produces quality software for Bar.");
+
+    action.createParam(PARAM_URL)
+      .setRequired(false)
+      .setDescription("URL of the organization.<br/> It must be less than 256 chars long.")
+      .setExampleValue("https://www.foo.com");
+
+    action.createParam(PARAM_AVATAR_URL)
+      .setRequired(false)
+      .setDescription("URL of the organization avatar.<br/> It must be less than 256 chars long.")
+      .setExampleValue("https://www.foo.com/foo.png");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    userSession.checkPermission(GlobalPermissions.SYSTEM_ADMIN);
+
+    String name = getAndCheckName(request);
+    String requestKey = getAndCheckKey(request);
+    String key = useOrGenerateKey(requestKey, name);
+    checkParamMaxLength(request, PARAM_DESCRIPTION, 256);
```

Comment:
they actually are part of the public documentation of the WS, so we should fail in the WS and ASAP and independently of the DB (column size could be larger and we still would have the check in the WS)
for the DB part, there checks are useless at DTO level as they are implemented as constraints in the DB